OpenRA.Mods.Common/Traits/Render/WithSpriteBody.cs

```diff
@@ -9,44 +9,83 @@
+		Requires<IBodyOrientationInfo>, Requires<RenderSpritesInfo>
 	{
 		[Desc("Animation to play when the actor is created.")]
 		public readonly string StartSequence = null;
 
-		[Desc("Animation to play when the actor is idle.")]
+		[Desc("Animation to play when the actor is idle/standing.")]
 		public readonly string Sequence = "idle";
 
-		public object Create(ActorInitializer init) { return new WithSpriteBody(init, this); }
+		public virtual object Create(ActorInitializer init) { return new WithSpriteBody(init, this); }
+
+		public virtual IEnumerable<IActorPreview> RenderPreviewSprites(ActorPreviewInitializer init, RenderSpritesInfo rs, string image, int facings, PaletteReference p)
+		{
+			var ifacing = init.Actor.Traits.GetOrDefault<IFacingInfo>();
+			var facing = ifacing != null ? init.Contains<FacingInit>() ? init.Get<FacingInit, int>() : ifacing.GetInitialFacing() : 0;
+
+			var anim = new Animation(init.World, image, () => facing);
+			anim.PlayRepeating(RenderSprites.NormalizeSequence(anim, init.GetDamageState(), Sequence));
+
+			yield return new SpriteActorPreview(anim, WVec.Zero, 0, p, rs.Scale);
+		}
+
+		public virtual int QuantizedBodyFacings(ActorInfo ai, SequenceProvider sequenceProvider, string race)
+		{
+			var rsi = ai.Traits.Get<RenderSpritesInfo>();
+			return sequenceProvider.GetSequence(rsi.GetImage(ai, sequenceProvider, race), Sequence).Facings;
+		}
 	}
 
-	class WithSpriteBody : UpgradableTrait<WithSpriteBodyInfo>, ISpriteBody
+	public class WithSpriteBody : UpgradableTrait<WithSpriteBodyInfo>, ISpriteBody
 	{
-		readonly Animation body;
-		readonly WithSpriteBodyInfo info;
+		public readonly Animation DefaultAnimation;
 
-		public WithSpriteBody(ActorInitializer init, WithSpriteBodyInfo info)
+		public WithSpriteBody(ActorInitializer init, WithSpriteBodyInfo info, Func<int> baseFacing)
 			: base(info)
 		{
-			this.info = info;
-
 			var rs = init.Self.Trait<RenderSprites>();
-			body = new Animation(init.Self.World, rs.GetImage(init.Self));
-			PlayCustomAnimation(init.Self, info.StartSequence, () => body.PlayRepeating(info.Sequence));
-			rs.Add(new AnimationWithOffset(body, null, () => IsTraitDisabled));
+
+			DefaultAnimation = new Animation(init.World, rs.GetImage(init.Self), baseFacing);
+			rs.Add(new AnimationWithOffset(DefaultAnimation, null, () => IsTraitDisabled));
+		}
+
+		public WithSpriteBody(ActorInitializer init, WithSpriteBodyInfo info)
+			: this(init, info, RenderSprites.MakeFacingFunc(init.Self))
+		{
+			if (Info.StartSequence != null)
+				PlayCustomAnimation(init.Self, Info.StartSequence,
+					() => DefaultAnimation.PlayRepeating(Info.Sequence));
+			else
+				DefaultAnimation.PlayRepeating(NormalizeSequence(init.Self, Info.Sequence));
+		}
+
+		public string NormalizeSequence(Actor self, string sequence)
+		{
+			return RenderSprites.NormalizeSequence(DefaultAnimation, self.GetDamageState(), sequence);
+		}
+
+		public void PlayCustomAnim(Actor self, string name)
```

Comment:
Just define a default value for `Action after = null`, add the sequence check (although I think we may prefer it to crash instead, as people really shouldn't be calling this with bogus sequences), and then change the callers to use the other method.